### PR TITLE
Fix EBPF issues.

### PR DESCRIPTION
**Pull request by komaljai**
Issues addressed -

1) In the parser generation remove an extraneous semicolon from file include code. 
2) In the parser generated code declare compiler_meta struct:
struct pna_global_metadata *compiler_meta__ = (struct pna_global_metadata *) skb->cb;
3) Make generated template file to have executable permissions
4) Generate template file by default (should have same name as pipeline)
5) Generate the introspection file by default with the SAME name as the pipeline.
6) Rename kfunc bpf_skb_p4tc_tbl_read to bpf_skb_p4tc_tbl_read
7) Move declaration of 'struct hdr_md' from post_parser.cpp to header.h file
8) Change post_parser.c file name to control_block.c
9) Remove '-i' and -'c' options.

### CODE REVIEW

File: backends/tc/backend.cpp [hunk 1]
```diff
@@ -114,28 +116,34 @@ bool Backend::ebpfCodeGen(P4::ReferenceMap *refMapEBPF, P4::TypeMap *typeMapEBPF
...
-    if (filename) progName = filename;
-    progName = progName.exceptLast(3);
-    progName = progName.trim("/\t\n\r");
     cstring parserFile = progName + "_parser.c";
     cstring postParserFile = progName + "_post_parser.c";
     cstring headerFile = progName + "_parser.h";
-    if (!options.cFile.isNullOrEmpty()) {
-        if (options.cFile.get(options.cFile.size() - 1) != '/') {
-            options.cFile = options.cFile + '/';
+    if (!options.outputFolder.isNullOrEmpty()) {
```
Comment (usha1830) [inline backends/tc/backend.cpp:137]: You can have this output folder check and update with / only once. need not be done for template and .c spearately

File: backends/tc/backend.cpp [hunk 2]
```diff
@@ -191,6 +199,8 @@ void ConvertToBackendIR::setPipelineName() {
     pipelineName = pipelineName.trim();
 }
 
+cstring ConvertToBackendIR::getPipelineName() { return pipelineName; }
```
Comment (usha1830) [inline backends/tc/backend.cpp]: In general, small functions like this which is only returning a member can be added to class declaration itself

**Review by usha1830 — APPROVED** [review]
Looks fine

File: backends/tc/backend.cpp (hunk 2 above)
Comment (komaljai) [reply backends/tc/backend.cpp]: resolved

File: backends/tc/backend.cpp (hunk 1 above)
Comment (komaljai) [reply backends/tc/backend.cpp:137]: resolved

**Review by usha1830 — APPROVED** [review]

File: backends/tc/backend.cpp [hunk 3]
```diff
@@ -114,28 +116,28 @@ bool Backend::ebpfCodeGen(P4::ReferenceMap *refMapEBPF, P4::TypeMap *typeMapEBPF
...
+    }
+    auto outstream = openFile(outputFile, false);
+    if (outstream != nullptr) {
+        *outstream << pipeline->toString();
+        outstream->flush();
+        std::filesystem::permissions(outputFile.c_str(),
+                                     std::filesystem::perms::owner_all |
+                                         std::filesystem::perms::group_all |
+                                         std::filesystem::perms::others_all,
+                                     std::filesystem::perm_options::add);
```
Comment (psivanup) [inline backends/tc/backend.cpp:132]: does the output file need execute permission as well?

**Review by psivanup — APPROVED** [review]
Looks fine to me

**Pull request closed without merging**